nemo/collections/nlp/modules/common/megatron/megatron_bert.py

```diff
@@ -49,52 +57,53 @@ class MegatronBertEncoder(BertModule):
     """
 
     def __init__(
-        self,
-        model_name,
-        vocab_file,
-        hidden_size=1024,
-        num_attention_heads=16,
-        num_layers=24,
-        max_seq_length=512,
-        tokenizer_type='BertWordPieceLowerCase',
-        init_method_std=0.02,
-        num_tokentypes=2,
+        self, model_name, config,
     ):
 
         super().__init__()
 
-        if not os.path.exists(vocab_file):
-            raise ValueError(f'Vocab file not found at {vocab_file}')
-
-        megatron_args = {
-            "num_layers": num_layers,
-            "hidden_size": hidden_size,
-            "num_attention_heads": num_attention_heads,
-            "max_position_embeddings": max_seq_length,
-            "tokenizer_type": tokenizer_type,
-            "vocab_file": vocab_file,
-        }
-
-        self.config = BertConfig(
-            num_layers=num_layers,
-            hidden_size=hidden_size,
-            num_attention_heads=num_attention_heads,
-            tokenizer_type=tokenizer_type,
-            vocab_file=vocab_file,
-        )
+        if not os.path.exists(config["vocab_file"]):
+            raise ValueError(f'Vocab file not found at {config["vocab_file"]}')
+
+        config['tokenizer_type'] = 'BertWordPieceLowerCase'
+        set_global_variables(extra_args_provider=None, args_defaults=config, ignore_unknown_args=True)
+        # read arguments back
+        args = get_args()
 
-        initialize_megatron(None, megatron_args, ignore_unknown_args=True)
-        init_method = init_method_normal(init_method_std)
 
-        self.language_model, self._language_model_key = get_language_model(
+        if args.distributed_backend == 'ddp':
+            set_model_parallel_rank(args.rank)
+            set_model_parallel_world_size(args.model_parallel_size)
+        else:
+            _initialize_distributed()
+
+        # Autoresume.
+        _init_autoresume()
+
```

Comment:
Why do we need it here?

Reply:
It was in megatron_initialize(). Should we remove it ?